src/main/java/org/spongepowered/api/item/inventory/Inventory.java

```diff
@@ -505,6 +505,35 @@
     InventoryArchetype getArchetype();
 
     /**
+     * Intersects the slots of both inventories.
+     * The resulting inventory will only contain slots
+     * that are present in both inventories.
+     *
+     * @param inventory the other inventory
+     * @return an inventory wrapping all slots that are present in both inventories
+     */
+    Inventory intersect(Inventory inventory);
+
+    /**
+     * Constructs a union of the slots in both inventories.
+     * The resulting inventory will contain all slots
+     * from both inventories.
+     *
+     * @param inventory the other inventory
+     * @return an inventory wrapping all slots of both inventories.
+     */
+    Inventory union(Inventory inventory);
+
+    /**
+     * Returns true if the given inventory is a descendant of this one.
+     * This includes deeply nested ones but not usually not query-results.
```

Comment:
Query Results are essentially a new Inventory View created to solve the query.

The Query Result returned, may, or may not be, a sub inventory of an existing inventory depending on the query.

If you have a reference to a sub-inventory, this will work in contains.